File: backend/src/main/java/ca/metricalsky/winston/entity/view/ReplyStatisticsView.java
package ca.metricalsky.winston.entity.view;

import java.time.OffsetDateTime;

public interface ReplyStatisticsView {

    String getCommentId();

    OffsetDateTime getCommentPublishedAt();

    Integer getCommentReplyCount();

    OffsetDateTime getCommentLastFetchedAt();

    Integer getFetchedReplyCount();

    OffsetDateTime getMostRecentReplyPublishedAt();
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/ReplyStatisticsView.kt
package ca.metricalsky.winston.database.view

import java.time.OffsetDateTime

interface ReplyStatisticsView {

    val commentId: String
    val commentPublishedAt: OffsetDateTime
    val commentReplyCount: Int
    val commentLastFetchedAt: OffsetDateTime
    val fetchedReplyCount: Int?
    val mostRecentReplyPublishedAt: OffsetDateTime?
}
